File: PSW-projekat/hospital-be-main/src/IntegrationLibrary/Core/Model/BloodRequest.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace IntegrationLibrary.Core.Model
{
    public class BloodRequest : BaseModel
    {
        
        [Required]
        public DateTime RequiredForDate { get; set; }
        [Required]
        public int BloodQuantity { get; set; }
        [Required]
        public string Reason { get; set; }
        [Required]
        public int DoctorId { get; set; }
        [Required]
        public RequestState RequestState { get; set; }
        [Required]
        public BloodType BloodType { get; set; }

        public int BloodBankId { get; set; }

        public BloodRequest(DateTime requiredForDat, int bloodQuantity, string reason, int doctorId, RequestState state, BloodType bloodType, int bloodBankId)
        {
            this.RequiredForDate = requiredForDat;
            this.BloodQuantity = bloodQuantity;
            this.Reason = reason;
            this.DoctorId = doctorId;
            this.RequestState = state;
            this.BloodType = bloodType;
            this.BloodBankId = bloodBankId;
        }
    }
}

File: PSW-projekat/PZL-Angular/src/app/modules/doctor-requests/model/blood-request.ts
import { BloodType } from "./blood-type";
import { DoctorBloodRequest } from "./doctor-blood-request";
import { RequestState } from "./request-state";

export class BloodRequest {
    id : number;
    requiredForDate : Date;
    bloodQuantity: number;
    reason: string;
    doctorId: number;
    requestState: RequestState;
    bloodType: BloodType;
    bloodBankId: number;
    comment: string;

    public constructor(obj?: any) {
        if (obj){
            this.id = obj.id;
            this.requiredForDate = obj.requiredForDate;
            this.bloodQuantity = obj.bloodQuantity;
            this.reason = obj.reason;
            this.requestState = obj.requestState;
            this.bloodType = obj.bloodType;
            this.bloodBankId= obj.bloodBankId;
            this.doctorId = obj.doctor.id;
            this.comment = obj.comment;
        }
    }
}
